.github/workflows/main.yml

```diff
@@ -0,0 +1,101 @@
+# Controls when the action will run.
+name: Fluxo para construcao e publicacao do container
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Versao do Tika a ser utilizada'
+        required: true
+        default: 1.25
+
+jobs:
+  printInputs:
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
+          echo "Tika v: ${{ github.event.inputs.version }}"
+          echo "Repo : ${{ secrets.REPO }}"
+  lint:
+    name: Lint do Dockerfile
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: lint
+        uses: luke142367/Docker-Lint-Action@v1.1.1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          target: Dockerfile
+  test:
+    name: Testando funcionamento do container
+    needs: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Build - Stack
+        run: docker-compose up -d
+
+      - name: Teste
+        run: docker run --network container:tika-server
+              appropriate/curl -s --retry 10
+              --retry-connrefused http://localhost:9998/
+
+  push:
+    name: Publica o Container
```

Comment:
Esse step só deveria acontecer caso o push fosse na main:

```
if: github.ref == 'refs/heads/main'
```

Reply:
Como vinculei ao workflow ele ficou como um step do processo, por isso que não separei. 
Mas vou alterar a parte toda para funcionar com o [push, pull_request]

Reply:
Mas veja que esse if ainda é necessário, mesmo com o on push e pull request, pq nesse passo do push só acontecerá quando for pra main e o workflow inteiro acontece no PR tb pra validar.

Reply:
Beleza, corrigido